cmd/ocagent: crash if OpenCensus receiver is not specified in configuration file

I was just creating a guide on the website for ocagent and while documenting the Zipkin receiver, I encountered an embarrasing crash with this configuration file below
```yaml
receivers:
    zipkin:
        address: "localhost:9411"
```
which gives
```shell
$ ./bin/ocagent_darwin --config zipkin.yaml 
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x38 pc=0x275f2ef]

goroutine 1 [running]:
github.com/census-instrumentation/opencensus-service/internal/config.(*Config).CanRunOpenCensusTraceReceiver(...)
	/Users/emmanuelodeke/Desktop/opencensus-service/internal/config/config.go:185
main.runOCReceiver(0xc0000eeb60, 0x4ca8070, 0x459e1c0, 0x3114a40, 0x459e1c0, 0x0, 0x0, 0x0)
	/Users/emmanuelodeke/Desktop/opencensus-service/cmd/ocagent/main.go:205 +0x41f
main.runOCAgent()
	/Users/emmanuelodeke/Desktop/opencensus-service/cmd/ocagent/main.go:99 +0x446
main.glob..func1(0x456cb20, 0xc00058f900, 0x0, 0x2)
	/Users/emmanuelodeke/Desktop/opencensus-service/cmd/ocagent/cli.go:35 +0x20
github.com/spf13/cobra.(*Command).execute(0x456cb20, 0xc0000e0010, 0x2, 0x2, 0x456cb20, 0xc0000e0010)
	/Users/emmanuelodeke/go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:766 +0x2ae
github.com/spf13/cobra.(*Command).ExecuteC(0x456cb20, 0x0, 0x0, 0x456cd80)
	/Users/emmanuelodeke/go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:852 +0x2c0
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/emmanuelodeke/go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:800
main.main()
	/Users/emmanuelodeke/Desktop/opencensus-service/cmd/ocagent/main.go:52 +0x2e
```

which steps from this
https://github.com/census-instrumentation/opencensus-service/blob/85ab29319ebdcf1b62faf4b24beb9cb66da61c24/internal/config/config.go#L184-L186

where on line 185, we didn't check that `c.Receivers.OpenCensus != nil`

--- a/internal/config/config.go
+++ b/internal/config/config.go
@@ -182,13 +182,15 @@ func (c *Config) OpenCensusReceiverCorsAllowedOrigins() []string {
 // CanRunOpenCensusTraceReceiver returns true if the configuration
 // permits running the OpenCensus Trace receiver.
 func (c *Config) CanRunOpenCensusTraceReceiver() bool {
-	return c != nil && c.Receivers != nil && !c.Receivers.OpenCensus.DisableTracing
+	return c != nil && c.Receivers != nil &&
+		c.Receivers.OpenCensus != nil && !c.Receivers.OpenCensus.DisableTracing
 }
 
 // CanRunOpenCensusMetricsReceiver returns true if the configuration
 // permits running the OpenCensus Metrics receiver.
 func (c *Config) CanRunOpenCensusMetricsReceiver() bool {
-	return c != nil && c.Receivers != nil && !c.Receivers.OpenCensus.DisableMetrics
+	return c != nil && c.Receivers != nil &&
+		c.Receivers.OpenCensus != nil && !c.Receivers.OpenCensus.DisableMetrics
 }
 
 // ZPagesDisabled returns true if zPages have not been enabled.

--- a/internal/config/config_test.go
+++ b/internal/config/config_test.go
@@ -61,3 +61,38 @@ exporters:
 		t.Errorf("Exporters.Zipkin.EndpointURL mismatch\nGot: %s\nWant:%s", g, w)
 	}
 }
+
+// Issue #377: If Config.OpenCensus == nil, invoking
+// CanRunOpenCensus{Metrics, Trace}Receiver() would crash.
+func TestOpenCensusTraceReceiverEnabledNoCrash(t *testing.T) {
+	// 1. Test with an in-code struct.
+	cfg := &config.Config{
+		Receivers: &config.Receivers{
+			OpenCensus: nil,
+		},
+	}
+	if cfg.CanRunOpenCensusTraceReceiver() {
+		t.Fatal("CanRunOpenCensusTraceReceiver: Unexpected True for a nil Receiver.OpenCensus")
+	}
+	if cfg.CanRunOpenCensusMetricsReceiver() {
+		t.Fatal("CanRunOpenCensusMetricsReceiver: Unexpected True for a nil Receiver.OpenCensus")
+	}
+
+	// 2. Test with a struct unmarshalled from a configuration file's YAML.
+	regressionYAML := []byte(`
+receivers:
+    zipkin:
+        address: "localhost:9410"`)
+
+	cfg, err := config.ParseOCAgentConfig(regressionYAML)
+	if err != nil {
+		t.Fatalf("Unexpected YAML parse error: %v", err)
+	}
+
+	if cfg.CanRunOpenCensusTraceReceiver() {
+		t.Fatal("yaml.CanRunOpenCensusTraceReceiver: Unexpected True for a nil Receiver.OpenCensus")
+	}
+	if cfg.CanRunOpenCensusMetricsReceiver() {
+		t.Fatal("yaml.CanRunOpenCensusMetricsReceiver: Unexpected True for a nil Receiver.OpenCensus")
+	}
+}

